src/Features/Core/Portable/ExternalAccess/UnitTesting/SolutionCrawler/UnitTestingWorkCoordinator.cs

```diff
@@ -59,34 +59,31 @@ internal sealed partial class UnitTestingWorkCoordinator
                 var entireProjectWorkerBackOffTimeSpan = UnitTestingSolutionCrawlerTimeSpan.EntireProjectWorkerBackOff;
 
                 _documentAndProjectWorkerProcessor = new UnitTestingIncrementalAnalyzerProcessor(
-                    listener, analyzerProviders, initializeLazily, _registration,
+                    listener, analyzerProviders, initializeLazily, Registration,
                     activeFileBackOffTimeSpan, allFilesWorkerBackOffTimeSpan, entireProjectWorkerBackOffTimeSpan, _shutdownToken);
 
                 var semanticBackOffTimeSpan = UnitTestingSolutionCrawlerTimeSpan.SemanticChangeBackOff;
                 var projectBackOffTimeSpan = UnitTestingSolutionCrawlerTimeSpan.ProjectPropagationBackOff;
 
-                _semanticChangeProcessor = new UnitTestingSemanticChangeProcessor(listener, _registration, _documentAndProjectWorkerProcessor, semanticBackOffTimeSpan, projectBackOffTimeSpan, _shutdownToken);
-
-                _registration.Workspace.WorkspaceChanged += OnWorkspaceChanged;
-                _registration.Workspace.TextDocumentOpened += OnTextDocumentOpened;
-                _registration.Workspace.TextDocumentClosed += OnTextDocumentClosed;
```

Comment:
instead of listening to events by hooking up to the workspace, we will instead get the events from teh new LegacySolutionEvents stream.